File: frontend/src/analysis/LocalAnalyzer.ts
export type LocalAnalysisInput={memoId:string;memoRevision:number;content:string;baseInstant:string;timeZone:string};
export interface LocalAnalyzer{analyze(input:LocalAnalysisInput):Promise<unknown>}
export class FakeLocalAnalyzer implements LocalAnalyzer{async analyze(input:LocalAnalysisInput){return{...input,analyzerVersion:'fake-v1'}}}

File: backend/src/main/java/local/personalmemo/analysis/domain/LocalAnalyzer.java
package local.personalmemo.analysis.domain;

import java.time.Instant;
import java.util.UUID;
import tools.jackson.databind.node.ObjectNode;

public interface LocalAnalyzer {
  String proposalSchemaVersion();

  AnalysisProvenance provenance();

  default String version() {
    return provenance().analyzerVersion();
  }

  ObjectNode analyze(
      UUID memoId, int revision, String content, Instant baseInstant, String timeZone);
}
